docs/source/guide/script_examples.md

```diff
@@ -0,0 +1,448 @@
+---
+title: Custom script examples
+short: Custom script examples
+tier: enterprise
+type: guide
+order: 0
+order_enterprise: 109
+section: "Create & Manage Projects"
+parent: "scripts"
+parent_enterprise: "scripts"
+date: 2024-07-30 13:31:47
+---
+
+The following examples work when [custom scripts](scripts) are enabled. 
+
+For details on implementing your own custom scripts, see [Label Studio Interface (LSI)](scripts#Label-Studio-Interface-LSI) and [Frontend API implementation details](scripts#Frontend-API-implementation-details). 
+
+## Plotly
+
+Use [Plotly](https://plotly.com/) to insert charts and graphs into your labeling interface. Charts are rendered in every annotation opened by a user. 
+
+Plotly should be loaded first from CDN: https://cdn.plot.ly/plotly-2.26.0.min.js. For security reasons, it's better to use a hash for script integrity. 
+
+
+![Screenshot of Plotly graph in Label Studio](/images/project/plotly.png)
+
+#### Script
+
+```javascript
+await LSI.import('https://cdn.plot.ly/plotly-2.26.0.min.js', 'sha384-xuh4dD2xC9BZ4qOrUrLt8psbgevXF2v+K+FrXxV4MlJHnWKgnaKoh74vd/6Ik8uF',);
+
+let data = LSI.task.data;
+if (window.Plotly && data) {
+  Plotly.newPlot("plot", [data.plotly]);
+}
+```
+
+**Related LSI instance methods:**
+
+* [import(url, integrity)](scripts#import-url-integrity)
+
+
+#### Labeling config
+
+You need to add `<View idAttr="plot"/>` into your config to render the Plotly chart. 
+
+For example:
+
+```xml
+<View>
+  <Text name="function" value="Is it increasing?" />
+  <Choices name="slope" toName="function">
+    <Choice value="Increasing" />
+    <Choice value="Decreasing" />
+    <Choice value="Non-monotonic" />
+  </Choices>
+  <View idAttr="plot"/>
+</View>
+```
+
+**Related tags:**
+
+* [View](/tags/view.html)
+* [Text](/tags/text.html)
+* [Choices](/tags/choices.html)
+* [Choice](/tags/choice.html)
+
+#### Data
+
+```json
+[
+  {
+    "plotly": {
+      "x": [1, 2, 3, 4],
+      "y": [10, 15, 13, 17],
+      "type": "scatter"
+    }
+  },
+  {
+    "plotly": {
+      "x": [1, 2, 3, 4],
+      "y": [16, 5, 11, 9],
+      "type": "scatter"
+    }
+  }
+]
+```
+
+## Custom validation
+
+In this example, the script checks to ensure that the annotation does not include obscenity or disallowed words. 
+
+The following script displays a modal if a user tries to submit an annotation with the word “hate” added to any audio transcription. 
```

Comment:
It may be worth clarifying that this implementation is a "soft block" where the user can submit if they attempt to do so again after seeing the modal once.